### PR TITLE
Ports - Jillianne

### PR DESCRIPTION
**Pull request by jillirami**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The model is linked to database and contains knowledge about the application, can be reused.
Describe in your own words what the Controller is doing in Rails | The controller processes requests and ties in the information of the model and the view for the application's overall function.
Describe in your own words what the View is doing in Rails | The view is visible to the user, related to HTML.
Describe an edge-case controller test you wrote | If variables passed to the update function were invalid, the expected status code was tested for.
What is the purpose of using strong params? (i.e. the params method in the controller) | Strong params prevent the user from directly updating model attributes, specifying the parameters information is passed.
How are Rails migrations related to Rails models? | Migrations establish and modify the parameters of the programs model.
Describe one area of Rails that are still unclear on | How to correctly use params in code and testing.

### CODE REVIEW

**Comment by tildeee** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | x
Answered comprehension questions | x
Successfully handles: Index, Show | x
Index & Show tests pass | x
Successfully handles: New, Create | x
New & Create tests pass | x
Successfully handles: Edit, Update | x
Tests for Edit & Update test valid & invalid task ids | x
Successfully handles: Destroy, Task Complete | x
Tests for Destroy & Task Complete include tests for valid and invalid task ids | x
Routes follow RESTful conventions | x
Uses named routes (like `_path`) | x
**Overall** |



Jillianne! Great work with this project! Your code meets all of the requirements I was looking for: Rails best practices, RESTful routing, and CRUD. Particularly, I loooove how well-written your controller tests are!


I only have a few notes for improvement. Otherwise, I think this is a great project submission.


Also, I want to note that your `verify` route and user experience is really cool, but above and beyond what we asked for! Let me know if you have any questions on that.


Well done!

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
+class TasksController < ApplicationController
+  # TASKS = [
+    # {name: "Call Ma", description: "Family First", completed_at: "", completion_date: ""}
+  # ]
```
Comment (tildeee) [inline app/controllers/tasks_controller.rb:4]: Don't forget to delete unused variables!

File: app/controllers/tasks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+  def index    
+    @tasks = Task.all.order(:created_at)
```
Comment (tildeee) [inline app/controllers/tasks_controller.rb:7]: Nice detail on ordering these

File: app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    if is_successful
+      redirect_to task_path(task.id)
+    end
+  end
+
+  def toggle_complete
+    task = Task.find_by(id: params[:id])
+
+    if task.completed_at == "Incomplete"
```
Comment (tildeee) [inline app/controllers/tasks_controller.rb:59]: Instead of assuming that `.completed_at` will be a string of either `"Incomplete"` or `"Complete"`, is there a better data type for this?

I think a boolean would be much better, and would prevent bugs. This way you don't need to do `if ... elsif ... else`, but just `if ... else`

Comment (tildeee) [inline app/controllers/tasks_controller.rb:59]: Is `completed_at` the best name for this attribute if it is describing the state of the task's completion?

File: app/views/tasks/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+<link href="stylesheets/index.css" rel="stylesheet" type="text/css">
```
Comment (tildeee) [inline app/views/tasks/index.html.erb:1]: You shouldn't need to do have this line in order to load your CSS. Rails should know how to load its own CSS

File: app/views/tasks/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    <h1>Jillianne's TaskList</h1>
+  </header>
+  <section>
+      <% if @tasks.length == 0 %>
+      <h3>No Tasks</h3>
+      <% else %>
+          <% @tasks.each do |task| %>
+              <br>
+              <td><%= task.name %></td>
+              <% if task.completed_at.nil? %>
```
Comment (tildeee) [inline app/views/tasks/index.html.erb:17]: You'll probably want to indent the code that's within the `if` (and also the `else`)

File: app/views/tasks/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+              <br>
```
Comment (tildeee) [inline app/views/tasks/index.html.erb:18]: Better to not use `br` tags as they're not semantic
